### PR TITLE
Implement IsPointInPath for Canvas

### PR DESCRIPTION
**Pull request by dzbarsky**
[Review on Reviewable](https://reviewable.io/reviews/servo/servo/8740)

### CODE REVIEW

**Comment by dzbarsky** [comment]
Depends on https://github.com/servo/rust-azure/pull/208

**Comment by frewsxcv** [comment]
servo/rust-azure#208 has merged.

**Comment by SimonSapin** [comment]
This probably needs `./mach cargo-update -p azure` as well. r=me with this.

@bors-servo delegate+

---

Reviewed 20 of 20 files at r1.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/servo/servo/8740#-:-K4cBCOwvzQO3xRmUMix:1800607500)*

**Comment by dzbarsky** [comment]
➜  servo git:(ispointinpath) ✗ ./mach cargo-update -p azure
components/servo
    Updating git repository `https://github.com/servo/rust-azure`
    Updating registry `https://github.com/rust-lang/crates.io-index`
Package `servo-glutin v0.4.3` does not have these features: `xlib, xf86vmode, xcursor`
ports/cef
    Updating git repository `https://github.com/servo/rust-azure`
    Updating registry `https://github.com/rust-lang/crates.io-index`
Package `servo-glutin v0.4.3` does not have these features: `xf86vmode, xlib, xcursor`
ports/gonk
    Updating git repository `https://github.com/servo/rust-azure`
    Updating registry `https://github.com/rust-lang/crates.io-index`
Package `servo-glutin v0.4.3` does not have these features: `xf86vmode, xcursor, xlib`

**Comment by SimonSapin** [comment]
Looks like the crate renames from #8757 are interfering. I reproduced this error locally, but it went away when I rebased on top of `upstream/master`.

**Comment by dzbarsky** [comment]
Looks like that worked, thanks

File: components/script/dom/canvasrenderingcontext2d.rs [hunk 1]
```diff
@@ -695,6 +695,18 @@ impl CanvasRenderingContext2DMethods for CanvasRenderingContext2D {
         self.ipc_renderer.send(CanvasMsg::Canvas2d(Canvas2dMsg::Clip)).unwrap();
     }
 
+    fn IsPointInPath(&self, x: f64, y: f64, fill_rule: CanvasFillRule) -> bool {
```
Comment (frewsxcv) [inline components/script/dom/canvasrenderingcontext2d.rs]: Spec link needed here

**Comment by KiChjang** [comment]
```
Ran 3728 tests finished in 869.0 seconds.
  • 3727 ran as expected. 707 tests skipped.
  • 1 tests had unexpected subtest results

Tests with unexpected results:
  ▶ Unexpected subtest result in /html/dom/interfaces.html:
  └ PASS [expected FAIL] CanvasRenderingContext2D interface: operation isPointInPath(unrestricted double,unrestricted double,CanvasFillRule)

  ▶ Unexpected subtest result in /html/dom/interfaces.html:
  └ PASS [expected FAIL] CanvasRenderingContext2D interface: operation isPointInPath(Path2D,unrestricted double,unrestricted double,CanvasFillRule)

  ▶ Unexpected subtest result in /html/dom/interfaces.html:
  └ PASS [expected FAIL] CanvasRenderingContext2D interface: document.createElement("canvas").getContext("2d") must inherit property "isPointInPath" with the proper type (42)

  ▶ Unexpected subtest result in /html/dom/interfaces.html:
  └ PASS [expected FAIL] CanvasRenderingContext2D interface: calling isPointInPath(unrestricted double,unrestricted double,CanvasFillRule) on document.createElement("canvas").getContext("2d") with too few arguments must throw TypeError

  ▶ Unexpected subtest result in /html/dom/interfaces.html:
  └ PASS [expected FAIL] CanvasRenderingContext2D interface: document.createElement("canvas").getContext("2d") must inherit property "isPointInPath" with the proper type (43)

  ▶ Unexpected subtest result in /html/dom/interfaces.html:
  └ PASS [expected FAIL] CanvasRenderingContext2D interface: calling isPointInPath(Path2D,unrestricted double,unrestricted double,CanvasFillRule) on document.createElement("canvas").getContext("2d") with too few arguments must throw TypeError
```
Looking good!

**Comment by dzbarsky** [comment]
Can I get r+/delegate+ please?

**Pull request merged**